src/state_machine/state.cpp

```diff
@@ -24,297 +24,306 @@ namespace hyped {
 
 namespace state_machine {
 
-State::State(Logger &log, Main *state_machine)
-    : log_(log),
-      data_(data::Data::getInstance()),
-      state_machine_(state_machine)
+//--------------------------------------------------------------------------------------
+//  General State
+//--------------------------------------------------------------------------------------
+
+State::State() : data_(data::Data::getInstance())
+{
+}
+
+void State::updateModuleData()
+{
+  embrakes_data_  = data_.getEmergencyBrakesData();
+  nav_data_       = data_.getNavigationData();
+  batteries_data_ = data_.getBatteriesData();
+  telemetry_data_ = data_.getTelemetryData();
+  sensors_data_   = data_.getSensorsData();
+  motors_data_    = data_.getMotorData();
+}
+
+//--------------------------------------------------------------------------------------
+//  Idling State
+//--------------------------------------------------------------------------------------
+
+Idling *Idling::instance_ = new Idling();
+
+State *Idling::checkTransition(Logger &log)
+{
+  updateModuleData();
+
+  State *next;
+  next = checkEmergencyStationery(log, embrakes_data_, nav_data_, batteries_data_, telemetry_data_,
```

Comment:
![image](https://blog.daraz.pk/wp-content/uploads/2019/03/25Mar19-5B.png)

I promise I'll also get to looking at the stuff you actually asked me to look at. :see_no_evil: :smile:

Reply:
That hurts... but I'll fix it anyway.